### DISCUSSION TITLE
Mac 12.7 install issues

### DISCUSSION THREAD

**Discussion by priestbh** [GeekFunkLabs/fluidpatcher#103] [General]
I am trying to install the fluidpatche desktop version my Macbookpro (prior to setting it up on a Pi zero 2 w) for a touch based bass midi controller I am wanting to make.

I have encountered the following issues with the mac instructions thus far:

1. fluid-soundfont-gm no longer exists as a homebrew package.  I searched and was able find this site https://member.keymusician.com/Member/FluidR3_GM/index.html and download a zip file to extract in to ~/fluidpatcher/SquishBox/sf2/ (before I saw https://github.com/GeekFunkLabs/fluidpatcher/discussions/36 about using archive.org)
2. pip3 is now just part of the python3 package (easy to work around)
3. now for my blocker: brew install ladspa-sdk is failing with a dependency error. According to https://formulae.brew.sh/formula/ladspa-sdk it appears the only depency is libsndfile (which I already have installed: libsndfile 1.2.2 is already installed and up-to-date.)

I haven't gotten to the patchcord compile yet due to the ladspa-sdk issue. 

Does anyone have a workaround for the  ladspa-sdk issue?

Also it may be time to update the mac install instructions.  Which I am happy to assist with once I get it working myself.

**Comment by priestbh** [comment]
I now see in the error and in the homebrew requirements ladspa-sdk is requiring Linux:
ladspa-sdk: Linux is required for this software.

Any workaround?

**Comment by priestbh** [comment]
Progress of a sort: I brushed off unused Makefile skills I haven't used in about 25 years.  I first tried to brew install ignoring dependencies.  That started the linker errors. Osx's dynamic linker /usr/bin/ld does not support the "--version-script" option.  I finally broke down and downloaded source from https://www.ladspa.org/ladspa_sdk/download.html  and removed "--version-script=gcc_exports.map" from the makefile targets. I finally got ladspa-sdk to compile and install in /usr/local/[include,lib,bin] and then found out the brew version of fluidsynth was not compiled with ladspa support. So tomorrow looks like I'll be compiling that from scratch as well.

**Reply by priestbh** [reply]
oh, one other error I had to fix:
applyplugin.c:9:10: fatal error: endian.h: No such file or directory
    9 | #include <endian.h>
https://stackoverflow.com/questions/27073837/fatal-error-endian-h-file-not-found
I had to change the include to be <machine/endian.h>

**Comment by albedozero** [comment]
I could definitely use some help updating the mac install instructions, and in general testing on Mac OS - help from anyone on this is most appreciated.

Some comments/clarification on your issues:

1. The included bank file bank1.yaml is based on using FluidR3_GM.sf2, but any GM (general midi) soundfont would work. Like fluidsynth, fluidpatcher doesn't require a specific soundfont - that choice is [up to the user](https://github.com/FluidSynth/fluidsynth/wiki/SoundFont). It is nice to have a good GM soundfont available. Maybe it makes sense to provide a link to download it - but as pointed out in FluidSynth/fluidsynth#479 there doesn't seem to be an official location. [Archive.org](https://archive.org/details/fluidr3-gm-gs) seems like a location that would be stable, but your suggestion or another might be good too.
2. Does the `brew install python3-pip` command fail if it's already installed, or is it just ignored? Any idea what minimum Mac OS version has pip installed by default?
3. ladspa-sdk is only needed to compile patchcord.c, and _that_ is only needed if you want to use LADSPA effect plugins to provide additional effects like delay, autowah, etc. on top of FluidSynth's built-in chorus and reverb. I have a bit of documentation about using them in bank files [here](https://github.com/GeekFunkLabs/fluidpatcher/blob/master/fluidpatcher/file_formats.md#ladspafx) (more to come soon I hope). Plugins have to be obtained separately and are platform-dependent. A bunch of good ones can be obtained in the debian/RPiOS packages `tap-plugins`, `swh-plugins`, and `wah-plugins` packages, and these might be available in brew or macports too. They are definitely _not_ available (in compiled form anyway) on Windows, and I've had issues gettting other LADSPA plugins to work on Windows, so it might be a problem with Mac too. The _L_ in LADSPA does stand for Linux, after all, so maybe it's not meant to be.

In summary, I think your issues are more with optional things and don't break fluidpatcher itself. Seems I should make it clearer what's optional and what's required in the install instructions. I do appreciate your efforts and continued comments/suggestions - keep it coming!
